### PR TITLE
CIV-5093 scheduler changes for Order Made End Date

### PR DESCRIPTION
**Pull request by deepthidoppalapudihmcts**
**Before creating a pull request make sure that:**

- [x ] commit messages are meaningful and follow good commit message guidelines
- [ x] README and other documentation has been updated / added (if needed)
- [x ] tests have been updated / new tests has been added (if needed)
- [ x] Screenshot attached for the Evidence of manual testing
- [x] Functional, Smoke and API tests have been run locally
- [x ] Full gradle build and run tests with coverage have been completed (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-5093

### Change description ###

Created a new scheduler to check the cases where the states is ORDER_MADE and its Judge approved end date is on that day
will trigger a new event for Work Allocation only for Applications where its type contains STAY_THE_CLAIM.  For this we have made following changes:
Addedd new task listener for "CheckStayOrderDeadlineEndTaskHandler" sechudler and publish the event "END_SCHEDULER_CHECK_STAY_ORDER_DEADLINE"for GA whose deadeline has reached. Update the case data variable esOrderProcessedByStayScheduler to "YES"

### Testing branch and instruction ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/GAOrderMadeTaskHandler.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.stream.Collectors;
+
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CHECK_ORDER_MADE_END_DATE;
+import static uk.gov.hmcts.reform.civil.enums.dq.GeneralApplicationTypes.STAY_THE_CLAIM;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@ConditionalOnExpression("${response.deadline.check.event.emitter.enabled:true}")
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/GAOrderMadeTaskHandler.java:23]: shouldnt this be different to response ?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckStayOrderDeadlineEndTaskHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    private List<CaseData> getOrderMadeCasesThatAreEndingToday() {
+        List<CaseDetails> orderMadeCases = caseSearchService.getGeneralApplications();
+        return orderMadeCases.stream()
+            .map(caseDetailsConverter::toCaseData)
+            .filter(caseData -> caseData.getJudicialDecisionMakeOrder().getJudgeApproveEditOptionDate() != null
+                && caseData.getGeneralAppType().getTypes().contains(STAY_THE_CLAIM)
+                && caseData.getJudicialDecisionMakeOrder().getEsOrderProcessedByStayScheduler().equals(YesOrNo.NO)
+                && (!now().isBefore(caseData.getJudicialDecisionMakeOrder().getJudgeApproveEditOptionDate()))
```
Comment (karthick-mohanasundaram-HMCTS) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckStayOrderDeadlineEndTaskHandler.java:53]: Please revisit the line 52 for field name @Azam-Hmcts

Comment (deepthidoppalapudihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckStayOrderDeadlineEndTaskHandler.java:53]: refactored code as per comments

File: src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/GAOrderMadeTaskHandler.java (hunk 1 above)
Comment (deepthidoppalapudihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/GAOrderMadeTaskHandler.java]: Modified the property name to judge-revisit-event-emitter

**Review by karthick-mohanasundaram-HMCTS — APPROVED** [review]

**Review by vasudevganesanhmcts — APPROVED** [review]

**Pull request merged**